File: src/DashboardUI/src/data-contracts/LocationDetails.ts
import { LocationWaterMeasurementDetails } from './LocationWaterMeasurementDetails';
import { DrawToolType } from './DrawToolType';

export interface LocationDetails {
  id: string;
  polygonWkt: string;
  drawToolType: DrawToolType;
  polygonAreaInAcres: number;
  additionalDetails: string | null;
  averageYearlyTotalEtInInches: number;
  averageYearlyTotalEtInAcreFeet: number;
  averageYearlyNetEtInInches: number | null;
  averageYearlyNetEtInAcreFeet: number | null;
  waterMeasurements: LocationWaterMeasurementDetails[];
}

File: src/API/WesternStatesWater.WestDaat.Common/DataContracts/LocationDetails.cs
namespace WesternStatesWater.WestDaat.Common.DataContracts;

public class LocationDetails
{
    public Guid Id { get; set; }

    public string PolygonWkt { get; set; } = null!;

    public DrawToolType DrawToolType { get; set; }

    public double PolygonAreaInAcres { get; set; }

    public string AdditionalDetails { get; set; } = null!;

    public double AverageYearlyTotalEtInInches { get; set; }

    public double AverageYearlyTotalEtInAcreFeet { get; set; }

    public double? AverageYearlyNetEtInInches { get; set; }

    public double? AverageYearlyNetEtInAcreFeet { get; set; }

    public LocationWaterMeasurementDetails[] WaterMeasurements { get; set; } = null!;
}
